game-app/game-core/src/main/java/games/strategy/engine/delegate/IDelegateBridge.java

```diff
@@ -94,4 +95,6 @@ public interface IDelegateBridge {
   GameData getData();
 
   void sendMessage(WebSocketMessage webSocketMessage);
+
+  ResourceLoader getResourceLoader();
```

Comment:
The bridge is the only object that allows delegates to receive outside data + the implementation of this interface `DefaultDelegateBridge` already holds a reference to a `ServerGame` instance where this method delegates this call to. So while the name might not be accurate just looking at the code layout it's pretty well fitting in my opinion.

Reply:
Hmm, I'm not sure I follow.

The 'delegate' pattern we have in TripleA is that the 'delegate' is both the local copy of an object and also a reference to every remote copy (if they exist). 'Getting' a resource loader is really a local activity, so it's odd to signal that local activity to each of the other remote instances.

Further though, there are a lot of no-op implementations which are significant code smells. Those smells tend to imply there should be a different interface involved.

Reply:
@DanVanAtta Regarding the no-op I tend to agree, however of you look closely we have like 3 no-op implementations apart from `DefaultDelegateBridge` that are used to simulate certain actions without actually performing changes to the underlying data. So yes, there's definitely some smell involved there but I don't think the resource loader is the issue.

Reply:
@DanVanAtta And I can't find any signs of remote access in this part of the code. In this case the bridge is used to provide delegates with arguments that are necessary for their implementation that can't be injected via the constructor (because of the architecture of delegate code). So even if a delegate is triggered over the network, the bridge would act local to the recipient only and their resource loader would provide the correct local files

Reply:
Any no-op implementation of an interface method is generally a smell. AFAIK that is fixed by breaking up the interface, so possibly this one method might need an interface if there is no other grouping. Hence, I disagree I think that 'resource-loader ' is causing that problem, it is adding to it I think. I don't have that strong of a opinion here that we should go gung-ho to fix this issue now, there are some trade-offs in life..

Re: delegates
Generally they are meant to be possible network calls, so it is odd to have something that does not fit that pattern and would also have a local effect. That's what I'm trying to point out.